### PR TITLE
Add CircleCI config to run TPU tests.

### PR DESCRIPTION
**Pull request by zcain117**
For every incoming commit, this PR will create a Docker image containing the commit's latest code and will run that Docker image on Google Kubernetes Engine on a TPU.

### CODE REVIEW

**Comment by zcain117** [comment]
`check_code_quality` failed with:

```
    torch from https://files.pythonhosted.org/packages/38/53/914885a93a44b96c0dd1c36f36ff10afe341f091230aad68f7228d61db1e/torch-1.6.0-cp36-cp36m-manylinux1_x86_64.whl#sha256=7669f4d923b5758e28b521ea749c795ed67ff24b45ba20296bc8cff706d08df8 (from transformers==3.0.2):
        Expected sha256 7669f4d923b5758e28b521ea749c795ed67ff24b45ba20296bc8cff706d08df8
             Got        36bbf4ab202de410d764b9156f3925b7d7037ad046f20690e576725a3826a2ac
```

I don't see how this latest commit could have caused this. I'll retry later

File: docker/transformers-pytorch-tpu/bert-base-cased.jsonnet [hunk 1]
```diff
@@ -0,0 +1,38 @@
+local base = import 'templates/base.libsonnet';
+local tpus = import 'templates/tpus.libsonnet';
+local utils = import "templates/utils.libsonnet";
+local volumes = import "templates/volumes.libsonnet";
+
+local bertBasedCase = base.BaseTest {
+  frameworkPrefix: "hf",
+  modelName: "bert-based-case",
```
Comment (LysandreJik) [inline docker/transformers-pytorch-tpu/bert-base-cased.jsonnet:8]: Shouldn't the `bert-based-case` and `bertBasedCase` be `bert-base-cased` and `bertBaseCased`?

By the way, why is this model specific? Are we using these values somewhere?

Comment (zcain117) [reply docker/transformers-pytorch-tpu/bert-base-cased.jsonnet]: Fixed that typo.

The `local bertBasedCase` is just a name of that jsonnet variable, it just needs to be a unique string but several punctuation marks are not allowed in variable names. We recommend 1 variable like this per test. On our team we have 1 file per "family" of tests, e.g. 1 file for pytorch resnet50 where the file contains [short test, long test] X [v2-8 TPU, v3-8 TPU, v100 GPU(s)] for the same model on the same ML framework. 

The `modelName` and `frameworkPrefix` are just used in generating the name of the GKE job. For example, a recent run was named: `hf-bert-based-case-example-v3-8-vtgrj`. Future runs will look like `hf-bert-base-cased-example-v3-8-xxxxx`

**Comment by zcain117** [comment]
See https://github.com/huggingface/transformers/pull/6219

**Pull request closed without merging**